trlx/sweep.py

```diff
@@ -1,65 +1,72 @@
-# Usage: python train_sweep.py --config configs/ray_tune_configs/ppo_config.yml --example-name ppo_sentiments
+# python -m trlx.sweep --config configs/sweeps/ppo_sweep.yml examples/ppo_sentiments.py
 import wandb
 import argparse
 from pathlib import Path
 
 import ray
 from ray.air import session
 from ray import tune
+import importlib
+import yaml
 
 import trlx
-from trlx.ray_tune import load_ray_yaml
 from trlx.ray_tune import get_param_space
 from trlx.ray_tune import get_tune_config
-from trlx.ray_tune import get_train_function
 from trlx.ray_tune.wandb import log_trials, create_report
 
 from ray.tune.logger import JsonLoggerCallback
 from ray.tune.logger import CSVLoggerCallback
 
 
-def tune_function(train_function, param_space: dict, tune_config: dict, resources: dict):
+def tune_function(
+    train_function, param_space: dict, tune_config: dict, resources: dict
+):
     tuner = tune.Tuner(
         tune.with_resources(train_function, resources=resources),
         param_space=param_space,
         tune_config=tune.TuneConfig(**tune_config),
-        run_config = ray.air.RunConfig(
-            local_dir="ray_results",
-            callbacks=[CSVLoggerCallback()]
+        run_config=ray.air.RunConfig(
+            local_dir="ray_results", callbacks=[CSVLoggerCallback()]
         ),
     )
 
     results = tuner.fit()
+    project_name = tune_config.get("project_name", "sweep")
 
     log_trials(
         tuner._local_tuner.get_experiment_checkpoint_dir(),
-        param_space["train"]["project_name"]
+        project_name,
     )
 
     create_report(
+        project_name,
         param_space,
         tune_config,
         Path(tuner._local_tuner.get_experiment_checkpoint_dir()).stem,
-        results.get_best_result().config
+        results.get_best_result().config,
     )
 
     print("Best hyperparameters found were: ", results.get_best_result().config)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
+    parser.add_argument("script", type=str, help="Path to the script")
     parser.add_argument(
-        "--example-name", type=str, default="ppo_sentiments", help="Name of the example"
-    )
-    parser.add_argument(
-        "--config", type=str, default=None, required=True, help="The config file defining the param_space."
+        "--config",
+        type=str,
+        required=True,
+        help="The config file defining the param_space.",
     )
     parser.add_argument(
         "--num-cpus", type=int, default=4, help="Number of CPUs to use per exp."
     )
     parser.add_argument(
```

Comment:
ray tune only supports one gpu right? Maybe remove this option for now

Reply:
yeah, even if I make it support more, this option would be redundant